support optional envvars for pod-infra-container

jdef opened the pull request:
This PR adds support for tweaking the environment variables of a pod's infra container, as discussed per #19881.

required for https://github.com/mesosphere/kubernetes-mesos/issues/739

/cc @dchen1107 @vishh

Commit:
support optional envvars for pod-infra-container

vishh commented on cmd/kubelet/app/server.go:
```diff
@@ -729,6 +729,7 @@ type KubeletConfig struct {
 
 	ExperimentalFlannelOverlay bool
 	NodeIP                     net.IP
+	DockerOptions              []dockertools.Option
```
nit: I'd name it as `ContainerRuntimeOptions` instead.

vishh commented:
Just one nit: I'd like this feature to generic to all runtimes. So let's avoid the Docker keyword except in the dockertools package. Otherwise LGTM. 
Thanks for the PR @jdef!

pmorie commented:
> Just one nit: I'd like this feature to generic to all runtimes. So let's avoid the Docker keyword except in the dockertools package.

+1

jdef commented:
Thanks for the fast review @vishh; I've addressed your comments, PTAL

On Fri, Feb 5, 2016 at 2:02 PM, Vish Kannan <notifications@github.com>
wrote:

> Just one nit: I'd like this feature to generic to all runtimes. So let's
> avoid the Docker keyword except in the dockertools package. Otherwise LGTM.
> Thanks for the PR @jdef <https://github.com/jdef>!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20719#issuecomment-180500876>
> .
>

jdef commented:
rebased

vishh commented on pkg/kubelet/dockertools/manager.go:
```diff
@@ -140,6 +142,21 @@ type DockerManager struct {
 
+// Option is a functional option type for DockerManager, useful for
+// completely optional settings.
+type Option func(*DockerManager)
```
nit: Can we pull this out of docker repo and place it in `pkg/kubelet/container` package? The goal was to make it cross-runtime compatible.

jdef replied:
done

vishh commented:
LGTM

jdef commented:
rebased
